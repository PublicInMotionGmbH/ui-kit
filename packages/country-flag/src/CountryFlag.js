import React from 'react'
import PropTypes from 'prop-types'

import { buildClassName } from '@talixo/shared'

import config from '../config'

/**
 * Component which represents Flag.
 *
 * @param {object} props
 * @param {string} [props.className]
 * @returns {React.Element}
 */
class CountryFlag extends React.PureComponent {
  componentDidMount () {
    if (this.context.registerFlag) {
      this.context.registerFlag()
    }
  }

  componentWillUnmount () {
    if (this.context.unregisterFlag) {
      this.context.unregisterFlag()
    }
  }

  render () {
    const { className, code, ...passedProps } = this.props

    return (
      <svg className={buildClassName('country-flag', className)} {...passedProps}>
        <use xlinkHref={`#${config.prefix}-${code.toLowerCase()}`} />
      </svg>
    )
  }
}

CountryFlag.contextTypes = {
  registerFlag: PropTypes.func,
  unregisterFlag: PropTypes.func
}

CountryFlag.propTypes = {
  /** Additional class name */
  className: PropTypes.string,

  /** Country code */
  code: PropTypes.string.isRequired
}

export default CountryFlag
