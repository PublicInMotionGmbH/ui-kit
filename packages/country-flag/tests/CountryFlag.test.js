import React from 'react'
import { shallow } from 'enzyme'

import CountryFlag from '../src/CountryFlag'

describe('<CountryFlag />', () => {
  it('renders flag correctly', () => {
    const wrapper = shallow(<CountryFlag code='pl' />)

    expect(wrapper).toMatchSnapshot()
  })

  it('should allow passing other props down', () => {
    const wrapper = shallow(<CountryFlag code='de' className='something' />)

    expect(wrapper).toMatchSnapshot()
  })

  it('should lower-case country code', () => {
    const wrapper = shallow(<CountryFlag code='PL' />)

    expect(wrapper).toMatchSnapshot()
  })
})
