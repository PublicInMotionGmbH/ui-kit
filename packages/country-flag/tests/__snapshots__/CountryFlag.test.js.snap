// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CountryFlag /> renders flag correctly 1`] = `
<svg
  className="talixo-country-flag"
>
  <use
    xlinkHref="#taf-pl"
  />
</svg>
`;

exports[`<CountryFlag /> should allow passing other props down 1`] = `
<svg
  className="talixo-country-flag something"
>
  <use
    xlinkHref="#taf-de"
  />
</svg>
`;

exports[`<CountryFlag /> should lower-case country code 1`] = `
<svg
  className="talixo-country-flag"
>
  <use
    xlinkHref="#taf-pl"
  />
</svg>
`;
